packages/firestore/src/core/sync_engine.ts

```diff
@@ -301,6 +317,21 @@ export class SyncEngine implements RemoteSyncer {
   applyRemoteEvent(remoteEvent: RemoteEvent): Promise<void> {
     this.assertSubscribed('applyRemoteEvent()');
 
+    // Update `receivedDocument` as appropriate for any limbo targets.
+    objUtils.forEach(remoteEvent.targetChanges, (targetId, targetChange) => {
+      const limboResolution = this.limboResolutionsByTarget[targetId];
+      if (limboResolution) {
+        if (
+          targetChange.addedDocuments.size > 0 ||
+          targetChange.modifiedDocuments.size > 0
+        ) {
+          limboResolution.receivedDocument = true;
+        } else if (targetChange.removedDocuments.size > 0) {
+          limboResolution.receivedDocument = false;
+        }
```

Comment:
Isn't there an `else` case here too? If we got a snapshot but there's no document at all it won't be added, modified, or removed. How are we supposed to handle that here?

If not, can you add a comment here/assertion about this?